network_plugin/public_nat.py

```diff
@@ -179,7 +180,24 @@ def nat_network_operation(vca_client, gateway, operation, rule_type, public_ip,
         function = gateway.add_nat_rule
         message = "Add"
         if _is_dnat(rule_type) and str(translated_port) == DEFAULT_SSH_PORT:
-            ctx.source.instance.runtime_properties['ssh_port'] = str(new_original_port)
+            retries_update = 3
+            update_pending = True
+            while retries_update > 0 and update_pending:
+                retries_update = retries_update - 1
+                try:
+                    ctx.source.instance.runtime_properties['ssh_port'] = str(
+                        new_original_port)
+                    ctx.source.instance.runtime_properties['ssh_public_ip'] =\
+                        public_ip
+                    ctx.source.instance.update()
```

Comment:
I saw an error about optimistic locking failure causing a retry after this change. currently publicNAT is not idempotent, so this cause ports/rules leaks. 
The update make the write to manager sooner which reduce the chance for updating an old version.

The better solution will be to:
1. understand who is competing with this node on the updates
2. Make publicNAT idempotent